### PR TITLE
Feature: select all cells with tags in toc

### PR DESCRIPTION
**Pull request by NPetz**
## References

closes [#8566](https://github.com/jupyterlab/jupyterlab/issues/8566)

## Code changes

- create a new method that selects all the cells that have all of the current selected tags and activates the first of those cells

## User-facing changes

- creates a new button the near the 'Clear Filters' button 

Screen Shot 2021-06-05 at 0 57 55

### CODE REVIEW

File: packages/toc/src/generators/notebook/tagstool/index.tsx [hunk 1]
```diff
@@ -214,6 +242,25 @@ class TagsToolComponent extends React.Component<IProperties, IState> {
         </span>
       );
     }
+    let command;
+
+    if (this.state.selected.length === 0) {
+      command = (
+        <span className={'toc-filter-button-na'}>
+          Select All Cells With Current Tags
```
Comment (goanpeca) [inline packages/toc/src/generators/notebook/tagstool/index.tsx:250]: Thanks for working on this @NPetz !

Could you please add internationalization to this ?

See for examples: https://github.com/jupyterlab/jupyterlab/blob/f0790b3ffd187b7dc923156b8ae227855abecefb/packages/toc/src/generators/notebook/toolbar_generator.tsx#L175

File: packages/toc/src/generators/notebook/tagstool/index.tsx [hunk 2]
```diff
@@ -214,6 +242,25 @@ class TagsToolComponent extends React.Component<IProperties, IState> {
...
+        </span>
+      );
+    } else {
+      command = (
+        <span
+          className={'toc-filter-button'}
+          onClick={this.selectAllCellsWithCurrentTags}
+        >
+          Select All Cells With Current Tags
```
Comment (goanpeca) [inline packages/toc/src/generators/notebook/tagstool/index.tsx:259]: Could you please add internationalization to this ?

See for examples: https://github.com/jupyterlab/jupyterlab/blob/f0790b3ffd187b7dc923156b8ae227855abecefb/packages/toc/src/generators/notebook/toolbar_generator.tsx#L175

**Review by goanpeca — CHANGES_REQUESTED** [review]
Thanks @NPetz 

Needs i18n 🌐  added? :)

**Comment by NPetz** [comment]
Thanks for the very quick review!
I added i18n, I hope it's fine!

**Review by blink1073 — APPROVED** [review]
Thank you!

**Comment by blink1073** [comment]
It looks like two things are required:

Please run `jlpm run lint` to pick up the auto-formatting changes.

Please copy `toc_show_tags.png` from the UI test [output](https://github.com/jupyterlab/jupyterlab/suites/2917486147/artifacts/65490703) and replace the file in `jupyterlab/ui-tests/reference-output/screenshots` locally to pick up the visual change.

**Comment by blink1073** [comment]
@mbektas, do you mind adding an entry to our contributor guide with the above suggestion for visual changes?

**Comment by NPetz** [comment]
> It looks like two things are required:
> 
> Please run `jlpm run lint` to pick up the auto-formatting changes.
> 
> Please copy `toc_show_tags.png` from the UI test [output](https://github.com/jupyterlab/jupyterlab/suites/2917486147/artifacts/65490703) and replace the file in `jupyterlab/ui-tests/reference-output/screenshots` locally to pick up the visual change.

@blink1073 

Thank you very much for your guidance!

I've run the linter and commited the changes, so that's fine.

However, I cannot find the directory you are talking about in my local files ( it seem to be listed in your .gitignore)... what's the proper course of action here?

**Comment by blink1073** [comment]
Whoops, that should have been `ui-tests` as the top level folder in your git checkout, so the same path otherwise.

**Comment by NPetz** [comment]
I'm sorry, I don't know why but when I forked and cloned the repo it didn't have the `ui-tests` directory in it.
I fetched from the upstream and  changed the screenshot file so now it should be ok.

**Comment by NPetz** [comment]
Well that's annoying... I guess I should have replaced `toggle-tag-2`  (and `toggle-tag-1` ?) too...
I don't understand why It didn't fail those tests the first time though....

**Comment by NPetz** [comment]
So I ran the ui-tests locally and replaced all the screenshots with the new button.
Hopefully it works this time?

**Comment by blink1073** [comment]
> Hopefully it works this time?

Hopefully!  Thanks for bearing with us, UI regression testing is a new feature, we're still working out the process.  ;)

**Comment by blink1073** [comment]
UI Test is green, thanks again!  This will be ready to merge once the other tests finish.

**Comment by NPetz** [comment]
> Hopefully! Thanks for bearing with us, UI regression testing is a new feature, we're still working out the process. ;)

No worries at all!
Thank you again for the support.

Hopefully I can continue contributing to the project.

**Comment by blink1073** [comment]
> Hopefully I can continue contributing to the project.

Great!

**Pull request merged**